### PR TITLE
Add event source mappings in Rulebook Activation form

### PR DESCRIPTION
**Pull request by lgalis**
First PR for source-event stream mapping for activations.
More changes and tests will follow in a separate PR.

Screenshot from 2024-08-12 08-17-43

Screenshot from 2024-08-12 08-18-53

Screenshot from 2024-08-12 08-19-15

Screenshot from 2024-08-12 08-19-26

### CODE REVIEW

**Review by tiyiprh — APPROVED** [review]
Looks great so far!

File: frontend/eda/rulebook-activations/RulebookActivationForm.tsx [hunk 1]
```diff
@@ -138,8 +144,8 @@ export function RulebookActivationInputs() {
   const { data: tokens } = useGet<EdaResult<AwxToken>>(
     edaAPI`/users/me/awx-tokens/?page=1&page_size=200`
   );
-  const { data: webhooks } = useGet<EdaResult<EdaWebhook>>(edaAPI`/webhooks/?page=1&page_size=200`);
 
+  const [_, setDialog] = usePageDialog();
```
Comment (marshmalien) [inline frontend/eda/rulebook-activations/RulebookActivationForm.tsx:148]: We're in the process of deprecating `usePageDialog` in favor of `usePageDialogs`. The new hook is capable of handling use cases where we have stacked dialogs. Could you please replace `usePageDialog` with `usePageDialogs` here?

File: frontend/eda/rulebook-activations/RulebookActivationForm.tsx [hunk 2]
```diff
@@ -215,7 +233,37 @@ export function RulebookActivationInputs() {
...
+            onClick={() =>
+              setDialog(
+                <SourceEventStreamMappingModal
+                  rulebook={rulebook}
+                  mappings={sourceMappings}
+                  setSourceMappings={setSourceMappings}
+                />
+              )
+            }
+          >{`${t('Manage event streams')}`}</Button>
```
Comment (marshmalien) [inline frontend/eda/rulebook-activations/RulebookActivationForm.tsx:250]: Small nitpick: We can remove the extra curly brackets and backticks here `{t('Manage event streams')}`

File: frontend/eda/rulebook-activations/components/SourceEventMapFields.tsx [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+      setValue(`mappings.${index}.webhook_name`, events.results[evIndex].name);
+    }
+  }
+
+  return (
+    <FormFieldGroupExpandable
+      isExpanded
+      header={
+        <FormFieldGroupHeader
+          titleText={{ text: t('Mapping ') + `${index + 1}`, id: `Mapping ${index}` }}
```
Comment (marshmalien) [inline frontend/eda/rulebook-activations/components/SourceEventMapFields.tsx:54]: We could translate this without concatenating with interpolation: ` t('Mapping {{count}}', { count: index + 1 })`

File: frontend/eda/rulebook-activations/components/SourceEventStreamMapping.tsx [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+              'filters intact. We swap out the webhook source type with a source of type ansible.eda.pg_listener.'
+          )}
+        </div>
+      }
+      variant={ModalVariant.large}
+      isOpen
+      onClose={onClose}
+      hasNoBodyWrapper
+    >
+      <ModalBoxBody style={{ padding: 0 }}>
```
Comment (marshmalien) [inline frontend/eda/rulebook-activations/components/SourceEventStreamMapping.tsx:125]: We could remove `ModalBoxBody` altogether if we don't want padding

File: frontend/eda/rulebook-activations/components/SourceEventStreamMapping.tsx [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+        <EdaPageForm onSubmit={onSubmit} submitText={t('Save')}>
```
Comment (marshmalien) [inline frontend/eda/rulebook-activations/components/SourceEventStreamMapping.tsx:126]: Could we add a Cancel button here?

**Pull request merged**